tensordict/memmap.py

```diff
@@ -298,13 +303,13 @@ def _init_shape(
         self._device = device
         self._shape = shape
         self._shape_indexed = None
-        self.transfer_ownership = transfer_ownership
         self.np_shape = tuple(self._shape)
         self._dtype = dtype
         self._ndim = len(shape)
         self._numel = prod(shape)
-        self._has_ownership = True
-        self._had_ownership = True
+        TRANSFER_OWNERSHIP[id(self.file)] = transfer_ownership
+        HAS_OWNERSHIP[id(self.file)] = True
+        HAD_OWNERSHIP[id(self.file)] = True
```

Comment:
I'm trying to work out if tying everything to the `id` of the file handle could result in duplication, and whether it's instead better to use a string of the absolute path?

I think we can probably get away with it because we only clean up the file if the user doesn't specify it and we create a temporary file for them. So for example this would be ok, even though the ids of `self.file` would differ

```python
mm1 = MemmapTensor(10, filename="myfile.memmap")
mm2 = MemmapTensor(10, filename="myfile.memmap")
```

Reply:
Should we use the filename instead?

Reply:
If it's easy enough to get the absolute filename then I think maybe yes. Just because if we change the behaviour for specified filenames in the future, we could end up with a race condition where one instance of `MemmapTensor` is not aware of another because they have different file handles pointing to the same file on disk.